### PR TITLE
Add current database / collection ID mapping

### PR DESCRIPTION
**Pull request by triozer**
### Description

Previously, when importing a Notion database with self-referencing relationships (records that reference other records within the same database), we had to follow a two-step process:
1.	First import the database
2.	Then perform a sync to populate the reference fields

With this PR, we're adding the current collection to the table mapping system during the import process itself, which allows us to handle circular references directly during the import. This eliminates the need for the additional sync step and ensures that self-referencing relationships are preserved correctly in a single operation.

This depends on https://github.com/framer/FramerStudio/pull/19792, which adds support for circular references when inserting items in a managed collection.

### Testing

- [ ] Importing a table with self-referencing values
  - [ ] Create a Notion database that contains records with relations to other records within the same database
  - [ ] Import this database into Framer
  - [ ] Verify that all self-referencing relationships are preserved correctly in the Framer collection

### CODE REVIEW

**Comment by tom-james-watson** [comment]
What's the context of this change?

**Comment by tom-james-watson** [comment]
Looking at this code again, it seems like I unnecessarily added this `const databaseIdKey = "notionDatabaseId"` when we already have `const pluginDatabaseIdKey = "notionPluginDatabaseId"` which seems to store the same thing. I think we can just consolidate these to use `pluginDatabaseIdKey` only?

**Comment by tom-james-watson** [comment]
Was the problem that `framer.getCollections` doesn't contain the current collection and that it's therefore missing from `databaseIdMap`?

File: plugins/notion/src/App.tsx [hunk 1]
```diff
@@ -32,6 +36,12 @@ export function AuthenticatedApp({ context }: AppProps) {
         })
     }, [databaseConfig])
 
+    useEffect(() => {
+        if (!databaseConfig) return
+
+        context.databaseIdMap.set(databaseConfig.id, context.collection.id)
+    }, [context, databaseConfig])
```
Comment (tom-james-watson) [inline plugins/notion/src/App.tsx:43]: Why is this needed?

**Review by tom-james-watson — COMMENTED** [review]
I'm still a bit confused about what this PR does. Why is this needed? Ideally there would be an associated issue that describes what the bug or task is. Mind adding a bit more of an explanation about what was not working before and why this is needed? 🙏

File: plugins/notion/src/App.tsx (hunk 1 above)
Comment (triozer) [reply plugins/notion/src/App.tsx:43]: image

It maps the selected Notion database to the current collection id. Could be better if moved to the `onSubmit`.

**Comment by tom-james-watson** [comment]
That description now is perfect :)

File: plugins/notion/src/App.tsx (hunk 1 above)
Comment (tom-james-watson) [reply plugins/notion/src/App.tsx:43]: Doesn't this

https://github.com/framer/plugins/pull/120/files/ffde46269b1e791e784d2afcf9650aeece4748b4#diff-0b64b774d3bc268fbeb7b15c667dac368a9c45ee00413a89778ec0efa2d4ef1fR784-R786

also do that, though?

Comment (triozer) [reply plugins/notion/src/App.tsx:43]: Ah, this would work only if `databaseId` is set which is not the case when importing a new one (importing vs syncing).

But you are right in the sense that we should never enter this if statement because when we're importing a new database, `databaseId` will be `null`, and when we're syncing an existing database, the ID will already be in the map from the collection iteration. So this is effectively dead code that can never be reached in either scenario.

Comment (tom-james-watson) [reply plugins/notion/src/App.tsx:43]: OK - best to remove then, it sounds like
